ops/main.py

```diff
@@ -335,6 +335,11 @@ def main(charm_class: ops.charm.CharmBase, use_juju_for_storage: bool = None):
 
     charm_state_path = charm_dir / CHARM_STATE_FILE
 
+    if use_juju_for_storage and not ops.storage._JujuStorageBackend.is_available():
```

Comment:
We shouldn't use storage's private attributes from main.

Reply:
fixed